ui/src/providers/SavedSets.js

```diff
@@ -1,98 +1,94 @@
-            //         ids
-            //       }
-            //     }`,
-            //     variables: {
-            //       sqon: sqon || {},
-            //       sort,
-            //     },
-            //   },
-            // });
-            this.setState({
-              loading: false,
-              sets: {
-                ...this.state.sets,
-                [setId]: { sqon, ids },
-              },
-            });
-            return { setId, ids };
-          },
-          fetchSets: async ({ sqon }) => {
-            this.setState({ loading: true, sets: this.state.sets });
-            const { data } = {};
-            // const { data } = await api({
-            //   endpoint: `/graphql`,
-            //   body: {
-            //     query: `query($sqon: JSON) {
-            //     sets {
-            //       hits(filters: $sqon) {
-            //         edges {
-            //           node {
-            //             ids
-            //             setId
-            //             sqon
-            //           }
-            //         }
-            //       }
-            //     }
-            //     }`,
-            //     variables: {
-            //       sqon,
-            //     },
-            //   },
-            // });
-            // const { sets } = data;
-            this.setState({
-              loading: false,
-              sets: {
-                ...this.state.sets,
-                // ...sets.hits.edges.reduce(
-                //   (acc, { node: { setId, ids, sqon } }) => ({ ...acc, [setId]: { ids, sqon } }),
-                //   {},
-                // ),
-              },
-            });
-          },
-          setSet: ({ setId, ids, sqon }) =>
-            this.setState({
-              sets: {
-                ...this.state.sets,
-                [setId]: { ids, sqon },
-              },
-            }),
-        }}
-        {...this.props}
-      />
-    );
-  }
-}
+  });
+
+  const { ARRANGER_API } = globals;
+  const { apiFetcher } = useDataContext({ apiUrl: ARRANGER_API, callerName: 'AggregationQuery' });
+  return (
+    <SavedSetsContext.Provider
+      value={{
+        state,
+        createSet: async ({ sqon, sort }) => {
+          setState({ loading: true, sets: this.state.sets });
+          const {
+            data: {
+              saveSet: { setId, ids },
+            },
+          } = await apiFetcher({ query: fetchSetsQuery, sqon });
+          // await api({
+          //   endpoint: `/graphql`,
+          //   body: {
+          //     query: `
+          //     mutation ($sqon: JSON!, $sort: [Sort]) {
+          //       saveSet(sqon: $sqon type: models path:"name" sort:$sort) {
+          //         sqon
+          //         setId
+          //         ids
+          //       }
+          //     }`,
+          //     variables: {
+          //       sqon: sqon || {},
+          //       sort,
+          //     },
+          //   },
+          // });
```

Comment:
is this cruft? or pending work?

Reply:
Pending work: https://github.com/nci-hcmi-catalog/portal/issues/1103
I'm going to review the feature branch for any commented features that were not covered in a ticket